core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -110,21 +110,32 @@ public void commitState(Snapshot snapshot)
       throws CommitException, UnknownTransactionStatusException {
     String id = snapshot.getId();
     try {
-      commitState(snapshot.getId());
-    } catch (CoordinatorException e) {
-      TransactionState state = abort(id);
-      if (state.equals(TransactionState.ABORTED)) {
-        rollbackRecords(snapshot);
-        throw new CommitException(
-            "committing state in coordinator failed. the transaction is aborted", e);
+      Coordinator.State state = new Coordinator.State(id, TransactionState.COMMITTED);
+      coordinator.putState(state);
+      logger.debug(
+          "transaction {} is committed successfully at {}", id, System.currentTimeMillis());
+    } catch (CoordinatorConflictException e) {
+      try {
+        Optional<Coordinator.State> s = coordinator.getState(id);
+        if (s.isPresent()) {
+          TransactionState state = s.get().getState();
+          if (state.equals(TransactionState.ABORTED)) {
+            rollbackRecords(snapshot);
+            throw new CommitException(
+                "committing state in coordinator failed. the transaction is aborted", e);
+          }
+        } else {
+          throw new UnknownTransactionStatusException(
+              "committing state failed with NoMutationException but the coordinator status doesn't exist",
```

Comment:
Just out of curiosity: is there any possible cause for this case?

Reply:
Basically, it's impossible unless something external deletes the status record. Honestly, in that case, I'm not sure we should throw `UnknownTransactionStatusException`, but I think it's okay as it's a very rare case.

Reply:
If an underlying database is somewhat broken, it could go to the state.
I think it's reasonable to mask the unexpected behavior with `UnknownTransactionStatusException`.